lib/theme_settings_object_validator.rb

```diff
@@ -28,12 +28,53 @@ def validate
 
   private
 
-  def validate_required_properties
+  def validate_properties
     @properties.each do |property_name, property_attributes|
-      if property_attributes[:required] && @object[property_name].nil?
-        @errors[property_name] ||= []
-        @errors[property_name] << I18n.t("themes.settings_errors.objects.required")
+      next if property_attributes[:required] && validate_required_property(property_name)
+      validate_property_type(property_attributes, property_name)
+    end
+  end
+
+  def validate_property_type(property_attributes, property_name)
+    value = @object[property_name]
+    type = property_attributes[:type]
+
+    return if (value.nil? && type != "enum")
+    return if type == "objects"
+
+    is_value_valid =
+      case type
+      when "string"
+        value.is_a?(String)
+      when "integer"
+        value.is_a?(Integer)
+      when "float"
+        value.is_a?(Float)
+      when "boolean"
+        [true, false].include?(value)
+      when "enum"
+        property_attributes[:choices].include?(value)
+      else
+        add_error(property_name, I18n.t("themes.settings_errors.objects.invalid_type", type:))
+        return
       end
+
+    if !is_value_valid
+      add_error(
+        property_name,
+        I18n.t("themes.settings_errors.objects.not_valid_#{type}_value", property_attributes),
+      )
     end
   end
+
+  def validate_required_property(property_name)
+    if @object[property_name].nil?
+      add_error(property_name, I18n.t("themes.settings_errors.objects.required"))
+    end
+  end
```

Comment:
```suggestion
  def validate_required_property(property_name)
    if @object[property_name].nil?
      add_error(property_name, I18n.t("themes.settings_errors.objects.required"))
      true
    else
      false
    end
  end
```

I believe this method returns an array of errors when a required property is not present which makes the condition for the `if` statement at line 33 evaluates to true and therefore skip the type validation step (skipping makes sense since the property not present)? If my understanding is correct, I think making this method return true/false explicitly would make things a bit easier to understand.

Reply:
Good point, I have made the changes 👍